src/brevitas/core/function_wrapper/clamp.py

```diff
@@ -73,3 +78,121 @@ def __init__(self, min_val: float) -> None:
     @brevitas.jit.script_method
     def forward(self, x: Tensor):
         return x.clamp_min(self.min_val)
+
+
+class FloatClamp(brevitas.jit.ScriptModule):
+    """"
+    ScriptModule for clamping minifloat formats to their inf/NaN implementations.
+
+    Currently, inf/NaN codes have to be encoded through the mantissa.
+    I.e. setting inf to 1101.111 (E4M3) is not a valid code.
+    """
+
+    __constants__ = [
+        'exponent_bit_width',
+        'mantissa_bit_width',
+        'exponent_bias',
+        'nan_values',
+        'inf_values',
+        'saturating']
+
+    def __init__(
+            self,
+            exponent_bit_width: Tensor,
+            mantissa_bit_width: Tensor,
+            exponent_bias: Tensor,
+            nan_values: Tuple[str],
+            inf_values: Tuple[str],
+            saturating: bool = False) -> None:
+        super(FloatClamp, self).__init__()
+
+        self.exponent_bit_width = exponent_bit_width
+        self.mantissa_bit_width = mantissa_bit_width
+        self.exponent_bias = exponent_bias
+
+        self.nan_values = nan_values if nan_values is not None else tuple()
+        self.inf_values = inf_values if inf_values is not None else tuple()
+        self.saturating = saturating
+
+        # check that NaN/inf values are all mantissa_bit_width long
+        if any(map(lambda x: len(x) > mantissa_bit_width, self.nan_values + self.inf_values)):
+            raise RuntimeError('NaN/inf codes need to be the same length as the mantissa.')
+
+        # inf without NaN not possible
+        if len(self.inf_values) == 0 and len(self.nan_values) == 0:
+            self.max_val_impl = StatelessBuffer(
+                max_float(self.exponent_bit_width, self.mantissa_bit_width, self.exponent_bias))
+        elif len(self.inf_values) > 0 and len(self.nan_values) == 0:
+            raise RuntimeError('Minifloat Error: inf value cannot exist without NaN value.')
```

Comment:
Check the error last, after exhausting all valid combinations, instead of having one valid combination as fallback